main/file.c

```diff
@@ -213,52 +285,193 @@ static void on_send(sub_send_cb_data_t *send_data){
 
         if(current_time_ms - sub_struct.last_tx_time_ms > sub_struct.tx_nack_timeout_ms){
 
-            if(xSemaphoreTake(sub_struct.mutex, portMAX_DELAY) == pdTRUE){
-
-                sub_struct.interval_ms = sub_struct.transmit_periods[SLEEP_MS];
-
-                xSemaphoreGive(sub_struct.mutex);
-
-            }
+            set_interval_ms(sub_struct.transmit_periods[SLEEP_MS]);
         }
     }
 }
 
 static void on_recive(sub_recv_cb_data_t *recv_data){
 
-    uint8_t obc_mac_addr[ESP_NOW_ETH_ALEN] = OBC_MAC_ADDRESS;
+    uint8_t mcb_mac_addr[ESP_NOW_ETH_ALEN] = MCB_MAC_ADDRESS;
+
+    if(memcmp(recv_data->dest_mac, sub_struct.dev_mac_address, ESP_NOW_ETH_ALEN)!=0){
```

Comment:
Sprawdzałeś jaki adres dostajesz gdy przychodzi wiadomość broadcast?

Reply:
czy ja dobrze rozumiem że jest jakiś specjalny adres mac odpowiedzialny za broadcast? to nie jest wysyłane poprostu do wszystkich pokolei?

Reply:
Chyba jest jakiś wybrany, w dokumentacji esp now powinno być z tego co pamiętam

Reply:
:0000